src/game/common/bezfwditerator.cpp

```diff
@@ -0,0 +1,106 @@
+/**
+ * @file
+ *
+ * @author xezon
+ *
+ * @brief Bezier Forward Iterator class.
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include "bezfwditerator.h"
+
+BezFwdIterator::BezFwdIterator()
+{
+    m_index = 0;
+    m_count = 0;
+    m_point.Zero();
+    m_add0.Zero();
+    m_add1.Zero();
+    m_add2.Zero();
+}
+
+BezFwdIterator::BezFwdIterator(int count, const BezierSegment *segment)
+{
+    m_point.Zero();
+    m_add0.Zero();
+    m_add1.Zero();
+    m_add2.Zero();
+    m_index = 0; // Thyme specific.
```

Comment:
reason for this comment?

Reply:
It is not set to 0 in original here, because it is not strictly required so.

Reply:
comment as a bugfix

Reply:
Comment removed. Is not really a bugfix. index is set to zero in start function, so its ok.